driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -86,12 +92,16 @@
     private final ConcurrentPool<UsageTrackingInternalConnection> pool;
     private final ConnectionPoolSettings settings;
     private final AtomicInteger generation = new AtomicInteger(0);
-    private final AtomicInteger lastPrunedGeneration = new AtomicInteger(0);
     private final ScheduledExecutorService sizeMaintenanceTimer;
     private ExecutorService asyncGetter;
     private final Runnable maintenanceTask;
     private final ConnectionPoolListener connectionPoolListener;
     private final ServerId serverId;
+    private final AtomicInteger numPinnedToCursor = new AtomicInteger(0);
+    private final AtomicInteger numPinnedToTransaction = new AtomicInteger(0);
```

Comment:
Done

Reply:
Can we use `PinnedStatsManager` instead of `PinnedStateManager`? Unlike `ServiceStateManager`, which does not contain statistics (`ServiceState.connectionCount` is used for correctness to avoid memory leaks), `PinnedStateManager` is nothing but a container for pinned connections statistics.